asv/benchmark.py

```diff
@@ -704,16 +705,22 @@ def __init__(self, name, func, attr_sources):
         self.unit = "bytes"
 
     def run(self, *param):
-        # We can't import asizeof directly, because we haven't loaded
-        # the asv package in the benchmarking process.
-        path = os.path.join(
-            os.path.dirname(__file__), 'extern', 'asizeof.py')
-        asizeof = importlib.machinery.SourceFileLoader('asizeof', path).load_module()
+        if ON_PYPY:
+            raise NotImplementedError("asizeof doesn't work on pypy")
+            return
+
+        try:
+            subprocess.check_output(['python', "-mpip", "install", 'pympler'])
```

Comment:
This works with all three environment backends, `conda`, `mamba` (as of #970) and `virtualenv` but is syntactically less than ideal. A better approach would be to modify the constraints of the `environment` class which is the base of all the backends.. Though YAGNI probably.